cleanlab/rank.py

```diff
@@ -152,14 +164,320 @@ def get_normalized_margin_for_each_label(
+) -> np.array:
+    """Returns the normalized entropy divided by "self-confidence".
+
+    Score is between 0 and 1.
+    1 - clean label (not an error).
+    0 - dirty label (label error).
+
+    Parameters
+    ----------
+    labels : np.array
+      A discrete vector of noisy labels, i.e. some labels may be erroneous.
+      *Format requirements*: for dataset with K classes, labels must be in {0,1,...,K-1}.
+
+    pred_probs : np.array (shape (N, K))
+      P(label=k|x) is a matrix with K model-predicted probabilities.
+      Each row of this matrix corresponds to an example x and contains the model-predicted
+      probabilities that x belongs to each possible class.
+      The columns must be ordered such that these probabilities correspond to class 0,1,2,...
+      `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
+
+    Returns
+    -------
+    confidence_weighted_entropy : np.array (float)
+      Return a score (between 0 and 1) for each example of its likelihood of
+      being correctly labeled. Assumes pred_probs is computed holdout/out-of-sample.
+
+    """
+
+    # Get confidence of given label
+    self_confidence = get_self_confidence_for_each_label(labels, pred_probs)
+
+    # Divide entropy by self confidence
+    confidence_weighted_entropy = get_entropy(**{"pred_probs": pred_probs}) / self_confidence
+
+    # Rescale
+    confidence_weighted_entropy = (
+        np.log(confidence_weighted_entropy + 1) / confidence_weighted_entropy
+    )
+
+    return confidence_weighted_entropy
+
+
+def get_entropy(pred_probs: np.array) -> np.array:
+    """Returns the normalized entropy of pred_probs.
+
+    Read more about normalized entropy here: https://en.wikipedia.org/wiki/Entropy_(information_theory)
+
+    Normalized entropy is used in active learning for uncertainty sampling: https://towardsdatascience.com/uncertainty-sampling-cheatsheet-ec57bc067c0b
+
+    Normalized entropy is between 0 and 1. Higher values of entropy indicate higher uncertainty in the model's prediction of the correct label.
+
+    Parameters
+    ----------
+    pred_probs : np.array (shape (N, K))
+      P(label=k|x) is a matrix with K model-predicted probabilities.
+      Each row of this matrix corresponds to an example x and contains the model-predicted
+      probabilities that x belongs to each possible class.
+      The columns must be ordered such that these probabilities correspond to class 0,1,2,...
+      `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
+
+    Returns
+    -------
+    entropy : np.array (float)
+
+    """
+
+    num_classes = pred_probs.shape[1]
+
+    # Note that dividing by log(num_classes) changes the base of the log which rescales entropy to 0-1 range
+    return -np.sum(pred_probs * np.log(pred_probs), axis=1) / np.log(num_classes)
+
+
+def subtract_confident_thresholds(labels: np.array, pred_probs: np.array) -> np.array:
```

Comment:
I think this is more of a helper method and not intended to be user-facing (will clutter the documentation IMO). 
So I'd prefer moving this method and other helper functions to a new module named something like `cleanlab/utils/label_quality_utils.py`
whose header says: `"""Helper functions for computing label quality scores"""`

That is unless you think beginner users would also find this `subtract_confident_thresholds()` method useful and it should be prominently displayed in the top-level documentation?

Reply:
I'm fine with either way. My only strong opinion is that subtract_confident_thresholds() should not be a user-facing method, it should either be private method `_subtract_confident_thresholds()` in this file as Curtis suggested, or buried deep inside utils/ like we currently have it.

Reply:
tl;dr
* change `subtract_confident_thresholds` to `_subtract_confident_thresholds`
* change `adj_pred_probs` to `adjust_pred_probs`
everywhere